src/orca/services/sevenbridges/tasks.py

```diff
@@ -1,17 +1,48 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from typing import Any, Optional
+from functools import wraps
+from typing import Any, Optional, cast
 
 from sevenbridges import Api
+from sevenbridges.models.enums import TaskStatus
 
+from orca.errors import OptionalAttrRequiredError, UnexpectedMatchError
 from orca.services.sevenbridges.client_factory import SevenBridgesClientFactory
 
 
+def project_required(method):
+    """Raise error if project is unset when calling method.
+
+    Args:
+        method: A SevenBridgesTasks method.
+
+    Returns:
+        A modified method that checks that the project attribute
+        is set before running the method.
+    """
+
+    @wraps(method)
+    def wrapper(self, *args, **kwargs):
+        if self.project is None:
+            name = method.__name__
+            message = f"`project` attribute must be set before calling `{name}()`."
+            raise OptionalAttrRequiredError(message)
+        return method(self, *args, **kwargs)
+
+    return wrapper
+
+
+# TODO: Should I change the class suffix to "Ops" to avoid confusion
```

Comment:
I had a moment of confusion about that when reviewing the other PR, so I would be in favor of this change.